docs/query-builder.txt

```diff
@@ -9,588 +9,1095 @@ Query Builder
+      :visible: false
+
+      [
+         { "title": "Kiss of the Spider Woman", ... },
+         { "title": "Spider-Man", ... },
+         { "title": "Spider-Man 2", ...},
+         ...
+      ]
+
+.. _laravel-query-builder-distinct:
+
+Retrieve Distinct Values
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+The following example shows how to use the ``distinct()``
+query builder method to retrieve all the different values
+of the ``year`` field for documents in the ``movies`` collections.
+
+.. literalinclude:: /includes/query-builder/MovieController.php
+   :language: php
+   :dedent:
+   :start-after: begin query distinct
+   :end-before: end query distinct
+
+.. _laravel-query-builder-aggregations:
+
+Aggregations
+~~~~~~~~~~~~
+
+The examples in this section show the query builder syntax you
+can use to perform aggregations, a set of calculations on the
+query results, to return the following information:
+
+- :ref:`Results grouped by common field values <laravel-query-builder-aggregation-groupby>`
+- :ref:`Count the of results <laravel-query-builder-aggregation-count>`
+- :ref:`Maximum value of a field <laravel-query-builder-aggregation-max>`
+- :ref:`Minimum value of a field <laravel-query-builder-aggregation-min>`
+- :ref:`Average value of a field <laravel-query-builder-aggregation-avg>`
+- :ref:`Summed value of a field <laravel-query-builder-aggregation-sum>`
+- :ref:`Aggregate matched results <laravel-query-builder-aggregate-matched>`
+
+.. _laravel-query-builder-aggregation-groupby:
+
+Results Grouped by Common Field Values Example
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The following example shows how to use the ``groupBy()`` query builder method
+to retrieve document data grouped by shared values of the ``runtime`` field.
+This example chains the following operations to match documents from the
+``movies`` collection that contain a ``rated`` value of ``G`` and include the
+``title`` field of one movie for each distinct ``runtime`` value:
+
+- Match only documents that contain a ``rated`` field value of "``G``" by
+  using the ``where()`` method
+- Group data by the distinct values of the ``runtime`` field, which is
+  assigned the ``_id`` field, by using the ``groupBy()`` method
+- Sort the groups by the ``runtime`` field by using the ``orderBy()`` method
+- Return ``title`` data from the last document in the grouped result by
+  specifying it in the ``get()`` method
+
+.. tip::
+
+   The ``groupBy()`` method calls the MongoDB ``$group`` aggregation operator
+   and ``$last`` accumulator operator. To learn more about these operators, see
+   :manual:`$group (aggregation) </reference/operator/aggregation/group/>`
+   in the {+server-docs-name+}.
+
+.. io-code-block::
+   :copyable: true
+
+   .. input:: /includes/query-builder/MovieController.php
+      :language: php
+      :dedent:
+      :start-after: begin query groupBy
+      :end-before: end query groupBy
+
+   .. output::
+      :language: json
+      :visible: false
```

Comment:
S: this example would be more illustrative if it produced a result where there were multiple titles per runtime.

Reply:
Thanks for the suggestion! I think it doesn't work that way since it applies the $last aggregation operator to fields that are not part of the aggregated _id as mentioned in the tip admonition, but perhaps @GromNaN can suggest a solution?